src/main/kotlin/net/atos/zac/flowable/cmmn/AanvullendeInformatieTaskListener.kt

```diff
@@ -60,36 +60,36 @@ class AanvullendeInformatieTaskListener : TaskListener {
             .planItemInstanceState("active")
             .list().size
 
-    private fun updateParentZaakStatus(planItemInstance: PlanItemInstance, statustypeOmschrijving: String) =
-        FlowableHelper.getInstance().let { flowableHelper ->
-            getZaak(planItemInstance).let { zaak ->
-                getStatustypeDescription(zaak).takeIf { it != statustypeOmschrijving }?.let {
-                    updateZaakStatus(zaak, statustypeOmschrijving)
-                }
+    private fun updateZaakStatusForTask(planItemInstance: PlanItemInstance, statustypeOmschrijving: String) =
+        getZaak(planItemInstance).let { zaak ->
+            getStatustypeDescription(zaak).takeIf { it != statustypeOmschrijving }?.let {
+                updateZaakStatus(zaak, statustypeOmschrijving)
             }
         }
 
     private fun getZaak(planItemInstance: PlanItemInstance) =
-        FlowableHelper.getInstance().let { flowableHelper ->
+        withFlowableHelper { flowableHelper ->
             flowableHelper.zaakVariabelenService.readZaakUUID(planItemInstance).let { zaakUUID ->
                 flowableHelper.zrcClientService.readZaak(zaakUUID)
             }
         }
 
     private fun getStatustypeDescription(zaak: Zaak): String =
-        FlowableHelper.getInstance().let { flowableHelper ->
+        withFlowableHelper { flowableHelper ->
             flowableHelper.zrcClientService.readStatus(zaak.status).let { zaakStatus ->
                 flowableHelper.ztcClientService.readStatustype(zaakStatus.statustype).omschrijving
             }
         }
 
     private fun updateZaakStatus(zaak: Zaak, statustypeOmschrijving: String) =
-        FlowableHelper.getInstance().let { flowableHelper ->
+        withFlowableHelper { flowableHelper ->
             LOG.info("Zaak with UUID '${zaak.uuid}': changing status to '$statustypeOmschrijving'")
             flowableHelper.zgwApiService.createStatusForZaak(
                 zaak,
                 statustypeOmschrijving,
                 STATUS_TOELICHTING
             )
         }
+
+    inline fun <T> withFlowableHelper(block: (FlowableHelper) -> T): T = block(FlowableHelper.getInstance())
```

Comment:
`with` suggests to me that `FlowableHelper.getInstance()` will be `this` in this context, so that should then be implemented as:
```suggestion
    inline fun <T> withFlowableHelper(block: () -> T): T = with(FlowableHelper.getInstance()) { block() }
```
or what do you think ?

Reply:
`with` does not actually work above as I would have expected. `this` in the lambda is still the one from calling function and not the `FlowableHelper`